### PR TITLE
feat(helm-chart/servicePrincipal): allow an existingSecret for holding armAuth.json

### PR DESCRIPTION
**Pull request by eyenx**
## Checklist
- [x] The title of the PR is clear and informative
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

As of today, when using the `servicePrincipal` method for achieving authentication, the `secretJSON` needs to be added as value in the chart.

With this change, the secret can already exist, which allows us to grab it from an external source like Azure Key Vault....

This way, the `secretJSON` value which is sensitive does not need to be added to the values and can be grabbed directly from an already existing Kubernetes Seecret.

## Fixes

Unable to deploy AGIC without exposing the `secretJSON` in the Helm values.

Signed-off-by: Toni Tauro <toni.tauro@adfinis.com>

### CODE REVIEW

**Comment by eyenx** [comment]
Any Updates on this?

**Review by akshaysngupta — DISMISSED** [review]

**Comment by akshaysngupta** [comment]
@eyenx can you share if you have tested this change ?

**Comment by eyenx** [comment]
@akshaysngupta I added unittests for this explicit example.

```
=== RUN   TestChart
=== RUN   TestChart/sample-config-empty
=== RUN   TestChart/sample-config-existing-secret
=== RUN   TestChart/sample-config-prohibited-target
=== RUN   TestChart/sample-config-workload-identity
=== RUN   TestChart/sample-config
--- PASS: TestChart (0.24s)
    --- PASS: TestChart/sample-config-empty (0.05s)
    --- PASS: TestChart/sample-config-existing-secret (0.04s)
    --- PASS: TestChart/sample-config-prohibited-target (0.05s)
    --- PASS: TestChart/sample-config-workload-identity (0.05s)
    --- PASS: TestChart/sample-config (0.05s)
PASS
ok      github.com/Azure/application-gateway-kubernetes-ingress/helm/ingress-azure/tests        (cached)
```

**Comment by eyenx** [comment]
Any Update @akshaysngupta

**Comment by eyenx** [comment]
Any Update @akshaysngupta ?

**Comment by eyenx** [comment]
Any Update @akshaysngupta ?

**Comment by eyenx** [comment]
Any Update @akshaysngupta ?

**Comment by ant-wtrog** [comment]
Hi @akshaysngupta, is there a chance to merge these changes in the near future? We're stuck at this point and would want to avoid coding a workaround.

Best Regards
Wolfgang

**Comment by eyenx** [comment]
Seems like Microsoft does not care about this. @akshaysngupta 

Not even a reaction if it is in scope or not. 

Really sad.

File: helm/ingress-azure/templates/deployment.yaml [hunk 1]
```diff
@@ -110,11 +110,16 @@ spec:
           path: /etc/kubernetes/
           type: Directory
       {{- if .Values.armAuth }}
-      {{- if eq .Values.armAuth.type "servicePrincipal"}}
+      {{- if and (eq .Values.armAuth.type "servicePrincipal") (not .Values.armAuth.existingSecret) }}
```
Comment (akshaysngupta) [inline helm/ingress-azure/templates/deployment.yaml:113]:
```yaml
      {{- if eq .Values.armAuth.type "servicePrincipal"}}
      - name: networking-appgw-k8s-azure-service-principal-mount
        secret:
          secretName: networking-appgw-k8s-azure-service-principal
      {{- else if .Values.armAuth.existingSecret }}
      - name: networking-appgw-k8s-azure-service-principal-mount
        secret:
          secretName: {{ .Values.armAuth.existingSecret }}
      {{- end}}
```

**Comment by akshaysngupta** [comment]
@eyenx sorry for delaying this PR so much. It looks perfectly fine. I have added a minor comment.

File: helm/ingress-azure/templates/deployment.yaml (hunk 1 above)
Comment (eyenx) [reply helm/ingress-azure/templates/deployment.yaml:113]: Thanks for your review.

Not sure this makes sense, the Auth Type will be servicePrincipal but we have an already existing secret, this way I would need to set the armAuthType to something else and set the existingSecret only. Or do I understand it wrong?

**Review by akshaysngupta — APPROVED** [review]

**Pull request merged**